src/components/answerOptions/index.jsx

```
@@ -45,33 +46,40 @@ const AnswerOptions = ({ birds, currentBird }) => {
-                    dispatch(setIsLevelCompleted(true));
-                    dispatch(setTotalScore(scoreOnTheLevel));
-                    currentUser.totalScoreForAllGames += scoreOnTheLevel;
-                    update();
-                    dispatch(setScoreOnTheLevel(MAX_SCORE_ON_THE_LEVEL));
-                  } else {
-                    click.classList.remove("won");
-                    if (!click.classList.contains("lost")) {
-                      const sound = new Audio(lostSound);
+              check()
+                .then(() => {
+                  dispatch(setSelectedBird(bird));
+                  const click = document.querySelector(`#${bird.name}`);
+                  if (!isLevelCompleted) {
+                    if (bird.name === currentBird.name) {
+                      click.classList.add("won");
+                      const sound = new Audio(wonSound);
                       sound.volume = SOUND_VOLUME;
                       sound.play();
-                      dispatch(setScoreOnTheLevel(scoreOnTheLevel - 1));
+                      dispatch(setIsCorrectCurrentBird(true));
+                      dispatch(setIsLevelCompleted(true));
+                      dispatch(setTotalScore(scoreOnTheLevel));
+                      currentUser.totalScoreForAllGames += scoreOnTheLevel;
+                      update();
+                      dispatch(setScoreOnTheLevel(MAX_SCORE_ON_THE_LEVEL));
+                    } else {
+                      click.classList.remove("won");
+                      if (!click.classList.contains("lost")) {
+                        const sound = new Audio(lostSound);
+                        sound.volume = SOUND_VOLUME;
+                        sound.play();
+                        dispatch(setScoreOnTheLevel(scoreOnTheLevel - 1));
+                      }
+                      click.classList.add("lost");
                     }
-                    click.classList.add("lost");
```

it should be a little different

it is not the client should check the token and whether the user is authorized, it's the server should return a 403 error if the user is not authorized

---

Done